### PR TITLE
Update Sabre to work with Sawtooth Rust SDK

### PR DESCRIPTION
**Pull request by agunde406**

### CODE REVIEW

File: sdk/src/lib.rs [hunk 1]
```diff
@@ -62,18 +62,138 @@ impl<'a> TpProcessRequest<'a> {
...
+    /// instead
+    ///
+    /// # Arguments
+    ///
+    /// * `addresses` - the addresses to fetch
+    fn get_state(&self, addresses: &[String]) -> Result<Vec<(String, Vec<u8>)>, WasmSdkError> {
+        self.get_state_entries(addresses)
+    }
+    /// get_state_entry queries the validator state for data at the
+    /// address given. If the  address is set, the data is returned.
```
Comment (jsmitchell) [inline sdk/src/lib.rs]: extra space

File: sdk/src/lib.rs [hunk 2]
```diff
@@ -62,18 +62,138 @@ impl<'a> TpProcessRequest<'a> {
...
 
-pub struct TransactionContext {}
+pub trait TransactionContext {
+    #[deprecated(
+        since = "0.2.0",
+        note = "please use `get_state_entry` or `get_state_entries` instead"
+    )]
+    /// get_state queries the validator state for data at each of the
+    /// addresses in the given list. The addresses that have been set
+    /// are returned. get_state is deprecated, please use get_state_entry or get_state_entries
```
Comment (jsmitchell) [inline sdk/src/lib.rs]: Thoughts on using #deprecated? https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-190-2016-05-26

Comment (agunde406) [reply sdk/src/lib.rs]: #deprecated is already used right above this comment

File: sdk/src/lib.rs (hunk 1 above)
Comment (agunde406) [reply sdk/src/lib.rs]: fixed

File: sdk/src/lib.rs (hunk 2 above)
Comment (jsmitchell) [reply sdk/src/lib.rs]: Hah! Maybe that makes the additional comments redundant. Thoughts?

Comment (agunde406) [reply sdk/src/lib.rs]: This currently matches what is in the rust sdk but I would be okay with removing the depreciated part of the doc comment if that is desired. This is what it looks like when docs are generated.
Screen Shot 2019-04-05 at 8 32 10 AM

Comment (jsmitchell) [reply sdk/src/lib.rs]: Ok, sounds good as is.

**Pull request merged**